whd-zephyr-integration/cyw43xxx_drv.c

```diff
@@ -0,0 +1,720 @@
+	}
+
+	whd_wifi_deregister_event_handler(cyw43xxx_ap_if, ap_event_handler_index);
+	whd_wifi_stop_ap(cyw43xxx_ap_if);
+	data->is_ap_up = 0;
+
+	k_sem_give(&data->sema_common);
+	return ret;
+}
+
+static int cyw43xxx_init(const struct device *dev)
+{
+	cy_rslt_t ret = CY_RSLT_SUCCESS;
+	struct cyw43xxx_wifi_runtime *data = (struct cyw43xxx_wifi_runtime *)dev->data;
+
+	cyhal_gpio_t sdio_cmd =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(WHD_SDIO_NODE, sdio_cmd_gpios);
+
+	cyhal_gpio_t sdio_clk =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(WHD_SDIO_NODE, sdio_clk_gpios);
+
+	cyhal_gpio_t sdio_d0 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS_IDX(WHD_SDIO_NODE, sdio_data_gpios, 0);
+
+	cyhal_gpio_t sdio_d1 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS_IDX(WHD_SDIO_NODE, sdio_data_gpios, 1);
+
+	cyhal_gpio_t sdio_d2 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS_IDX(WHD_SDIO_NODE, sdio_data_gpios, 2);
+
+	cyhal_gpio_t sdio_d3 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS_IDX(WHD_SDIO_NODE, sdio_data_gpios, 3);
+
+	ret = cyhal_sdio_init(&sdio_obj, sdio_cmd, sdio_clk, sdio_d0, sdio_d1, sdio_d2, sdio_d3);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("cyhal_sdio_init failed ret = %d \r\n", ret);
+	}
+
+	k_tid_t tid = k_thread_create(
+		&cyw43xxx_wifi_event_thread, cyw43xxx_wifi_event_stack,
+		CONFIG_CYW43XXX_WIFI_EVENT_TASK_STACK_SIZE, (k_thread_entry_t)cyw43xxx_event_task,
+		NULL, NULL, NULL, CONFIG_CYW43XXX_WIFI_EVENT_TASK_PRIO, K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_name_set(tid, "cyw43xxx_event");
+
+	ret = cybsp_wifi_init_primary_extended(&cyw43xxx_if, NULL, NULL, NULL, NULL);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("cybsp_wifi_init_primary_extended failed ret = %d \r\n", ret);
+	}
+	ret = whd_management_set_event_handler(cyw43xxx_if, sta_link_events, link_events_handler,
+					       NULL, &sta_event_handler_index);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("whd_management_set_event_handler failed ret = %d \r\n", ret);
+	}
+
+	k_sem_init(&data->sema_common, 1, 1);
+	k_sem_init(&data->sema_scan, 0, 1);
+
+	return ret;
+}
+
+/* Implement WHD memory wrappers */
+void *whd_mem_malloc(size_t size)
+{
+	return k_malloc(size);
+}
+
+void *whd_mem_calloc(size_t nitems, size_t size)
+{
+	return k_calloc(nitems, size);
+}
+
+void whd_mem_free(void *ptr)
+{
+	k_free(ptr);
+}
+
+static const struct wifi_mgmt_ops cyw43xxx_wifi_mgmt = {
+	.scan = cyw43xxx_mgmt_scan,
+	.connect = cyw43xxx_mgmt_connect,
+	.disconnect = cyw43xxx_mgmt_disconnect,
+	.ap_enable = cyw43xxx_mgmt_ap_enable,
+	.ap_disable = cyw43xxx_mgmt_ap_disable,
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+	.get_stats = cyw43xxx_mgmt_wifi_stats,
+#endif
+
+};
+
+static const struct net_wifi_mgmt_offload cyw43xxx_api = {
+	.wifi_iface.iface_api.init = cyw43xxx_mgmt_init,
+	.wifi_iface.send = cyw43xxx_mgmt_send,
+	.wifi_mgmt_api = &cyw43xxx_wifi_mgmt,
+};
+
+NET_DEVICE_DT_INST_DEFINE(0, cyw43xxx_init, NULL, &cyw43xxx_wifi_data, NULL,
+			  CONFIG_WIFI_INIT_PRIORITY, &cyw43xxx_api, ETHERNET_L2,
+			  NET_L2_GET_CTX_TYPE(ETHERNET_L2), NET_ETH_MTU);
```

Comment:
whether it should be WHD_LINK_MTU instead of NET_ETH_MTU?
we have seen that in case of complete buffer usage the WHD is not able to handle the incoming packet?
what we have seen that the value of NET_ETH_MTU is 1514 where as WHD_LINK_MTU is 1544.

Reply:
@npal-cy can you share your thoughts?

Reply:
Hi @prayassamriya , yes it should use WHD_LINK_MTU, which include addition bytes for header. I will update.

Reply:
Thanks you